### PR TITLE
AGDIGGER-60 - fetching and saving artifacts api

### PR DESCRIPTION
**Pull request by wtrocki**
# Motivation

Fetch and save artifacts API.
PR against master - previous PR was created against @aliok branch.
For old comments see: https://github.com/aliok/digger-java/pull/1

review @matzew @aliok

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,11 +4,21 @@
...
     <packaging>jar</packaging>
 
+  <properties>
+    <junit.version>4.11</junit.version>
+    <mockito-core.version>1.9.5</mockito-core.version>
+    <jenkins.client.version>0.3.7-SNAPSHOT</jenkins.client.version>
+    <jtwig.templates.version>5.65</jtwig.templates.version>
+    <slf4j.api.version>1.7.21</slf4j.api.version>
+    <slf4j-log4j12.version>1.7.21</slf4j-log4j12.version>
+    <assertj-core.version>3.6.1</assertj-core.version>
```
Comment (matzew) [inline pom.xml:19]: :heart_eyes:

File: src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  private ArtifactsService artifactsService;
+
+  @Before
+  public void beforeTests() {
+    artifactsService = new ArtifactsService();
+  }
+
+  @Test
+  public void shouldFetchArtifactsWithNoResults() throws Exception {
+    JobWithDetails job = mock(JobWithDetails.class);
```
Comment (josemigallas) [inline src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java:38]: I would gather all mocked objects together, below @Mock annotation or at least at the top of the test.

File: src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ArtifactsServiceTests {
+
+  @Mock
+  private JenkinsServer server = mock(JenkinsServer.class);
```
Comment (josemigallas) [inline src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java]: No need to instantiate mocks when using @Mock annotation.

File: src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java (hunk 2 above)
Comment (matzew) [reply src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java:38]: @josemigallas how about we send a follow request to turn the `mock` static calls into proper annotations ? On a new PR ?

Comment (josemigallas) [reply src/test/java/com/redhat/digkins/services/ArtifactsServiceTests.java:38]: My point is about consistency, some mocks are under @Mock, some others don't. But yes, whatever suits the your workflow better.

**Comment by wtrocki** [comment]
Updated all comments. Change was also reviewed in previous PR. Going to merge now!

File: src/main/java/com/redhat/digkins/Main.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.redhat.digkins;
```
Comment (aliok) [inline src/main/java/com/redhat/digkins/Main.java]: -1

**Pull request merged**